converters/fromZigbee.js

```diff
@@ -30,7 +31,11 @@ const tuyaGetDataValue = (dataType, data) => {
     case common.TuyaDataTypes.value:
         return utils.convertMultiByteNumberPayloadToSingleDecimalNumber(data);
     case common.TuyaDataTypes.string:
-        return data.map((byte) => String.fromCharCode(byte)).join('');
+        // If we do the old map, for some reason hex-letters don't work.
```

Comment:
Why doesn't this work? Code should do the same. 

Try to debug by changing to  

```
let dataString = data.map((byte) => String.fromCharCode(byte)).join('');
console.log(dataString);
return dataString
```

Reply:
I 100% agree that the code should do the same, but it gave me incorrect strings:

For some reason whenever I did the map some bytes wouldn't correctly transform into their char counterpart.
0x30 -> 0x39 all converted to '0' -> '9'. That's good.
But when there were characters 0x61 -> 0x66 (which are 'a' -> 'f') there was some weirdness going on where they all became '0'.

Reply:
I see, then it's ok if you move `let dataString = '';` right above the for loop.

Reply:
That gives me a lint issue. That says variables should not be declared in switch cases.

Reply:
In that case, is it possible to replace `let dataString = '';` with:
```js 
const bytesToString = (data) => {
  let dataString = '';
  for (let i = 0; i < data.length; ++i) {
    dataString += String.fromCharCode(data[i]);
   }
   return dataString;
}
```

and in the case just call with `return bytesToString(data)`

Reply:
Forget the above comment, move  `let dataString = '';` to the case and above put `// eslint-disable-next-line`.